### PR TITLE
shuffle the audio queue locally

### PR DESCRIPTION
**Pull request by mueslimak3r**
**Changes**

* allow `mediaManager` to manage shuffling a new audio queue. This means clicking "shuffle (all)" will create a queue with the shuffle button enabled, and toggling it off will give the user an un-shuffled queue.

* gave the relevant `playNow()` methods a `shuffle` (boolean) parameter

* bypassed `getItemsToPlay()` handling shuffling audio items. This could be done more cleanly by having it internally ignore the `shuffle` parameter for audio

* fixed the query sorting of MusicArtist so songs are sorted by album and are in order within the albums

**Issues**
* starting audio playback by using the "shuffle (all)" buttons didn't allow for truly un-shuffling
* clicking "play all" on a music artist would sometimes (but not always) queue the songs unordered

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java [hunk 1]
```diff
@@ -193,7 +193,7 @@ public void onResponse(ItemsResult response) {
                                     TvApp.getApplication().getCurrentActivity().startActivity(intent);
                                     break;
                                 case "Audio":
-                                    mediaManager.playNow(Arrays.asList(response.getItems()), startIndex);
+                                    mediaManager.playNow(Arrays.asList(response.getItems()), startIndex, false);
```
Comment (DavidFair) [inline app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java:196]: For these values could we use a named bool so future devs don't have to dive into the docs?

```kotlin
shuffle = false
mediaManager.playNow(Arrays.asList(response.getItems()), startIndex, shuffle);
```

I find it prevents code from gradually turning into this:

```kotlin
mediaManager.foo(response, true, false);
```

File: app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsActivity.java [hunk 2]
```diff
@@ -1599,15 +1599,16 @@ public void onResponse(UserItemDataDto response) {
     }
 
     protected void play(final BaseItemDto item, final int pos, final boolean shuffle) {
+        boolean shuffleQuery = item.getBaseItemType() == BaseItemType.MusicAlbum || item.getBaseItemType() == BaseItemType.MusicArtist || item.getBaseItemType() == BaseItemType.Audio ? false : shuffle;
```
Comment (DavidFair) [inline app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsActivity.java:1602]: Can this be moved into it's own private method? That way the logic can be broken down and documented without causing `play` to grow

File: app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java (hunk 1 above)
Comment (mueslimak3r) [reply app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java:196]: I'm definitely open to this.
I do think this accommodation is only useful in IDEs without syntax highlighting / introspection features.
In android studio, parameters like this have their name shown beside them.

Comment (DavidFair) [reply app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java:196]: Ah feel free to disregard this 👍 
I tend to use both vim and the IDE so this is why I spotted it intislly.
But I'm also not against either jumping to the source or using the proper tool compared to adding noise to the source code. Plus multiple booleans is indicative of a hidden dataclass we can pull out

File: app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsActivity.java (hunk 2 above)
Comment (mueslimak3r) [reply app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsActivity.java]: I went ahead and removed all the uses of `shuffleQuery`.
Instead I just removed the use of `shuffle` for creating the queries for `MusicAlbum` & `MusicArtist` in `getItemsToPlay`. If shuffling audio queues locally is accepted, leaving `shuffle` in as an option for the queries, and working around it externally, is messy.

File: app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java (hunk 1 above)
Comment (mueslimak3r) [reply app/src/main/java/org/jellyfin/androidtv/data/eventhandling/TvApiEventListener.java:196]: I'm a fan of vim too
Have you tried plugins like this:
https://github.com/uiiaoo/java-syntax.vim

I just tested with neovim and it does add nice colors to the code but it still doesn't provide the parameter names for methods.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**